### DISCUSSION TITLE
i am having this issue when i scan device with .details function it doesnt show me its detail

### DISCUSSION THREAD

**Discussion by roysiki** [hbldh/bleak#1448] [Q&A]
it is the code i m running 

```python
import asyncio
from bleak import BleakScanner

async def main():
    devices = await BleakScanner.discover()
    for d in devices:
        print(d.details)

asyncio.run(main())
```

and i dont know its not showing the detail though if i just go with d it shows me the device name and other stuff but with details it showing me something else i attached the output. though i m new to it.. i shall be very thankfull to you 
![error2222](https://github.com/hbldh/bleak/assets/148053717/3190bf0c-d2ff-48b3-9c99-588a36fb4171)

**Comment by dlech** [comment]
`details` is for internal use. Try the version of `discover()` with `return_adv=True` and have a look at the returned advertisement data.

**Reply by roysiki** [reply]
hi thanks for replying firstly .. i have tried it and i am attaching the screenshot of it it scan the device around me 
![error 33](https://github.com/hbldh/bleak/assets/148053717/8606a8f5-361b-4a50-a790-dedd3926af2b)
but i want more information as i was reading in some blog i will attached the screen shot of it  ...
![Capture44](https://github.com/hbldh/bleak/assets/148053717/ce9c43b5-e710-43da-973b-3c8df70f2d10)
plus this command works on linux and mac and idk why it giving me some odd output when i run this

**Reply by dlech** [reply]
What version of Bleak are you using?

**Reply by roysiki** [reply]
bleak==0.21.1

**Reply by dlech** [reply]
It would make it easier if you copied and pasted the actual text of the code instead of screenshots.

You need to change the line `for d in devices` to `for d, a in devices.values()` and then inspect `a` (the advertising data).
